### PR TITLE
Add in-game stickyban interface

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Adds an in-game stickyban interface, by MrStonedOne: https://github.com/tgstation/-tg-station/pull/11205.
2) Admins will no longer receive a message if there's no memos when they join.
3) Ports /tg/'s admin bans: these bans will actually apply to admins and prevent them from joining the server. Every admin can add a single adminban to the database to prevent abuse.
4) Reload admins/de-admin will now properly remove proc call verbs.

*Note: Please ensure "Security" in Dream Daemon is set to "Trusted", or this will not work.*

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
OH MY GOSH IT'S CHRISTMAS

**Comment by Earthdivine** [comment]
Yes yes yes merge now fuck conflicts

**Comment by Regen1** [comment]
:+1: :100:

**Comment by SamCroswell** [comment]
Mark.
Mark please.
Marry me.

**Comment by Markolie** [comment]
All seems well.

File: code/modules/admin/stickyban.dm [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		. += "<li><a href='?_src_=holder;stickyban=remove_alt&ckey=[ckey]&alt=[ckey(key)]'>\[-\]</a>[key]</li>"
+	. += "</ol>\n"
+
+/datum/admins/proc/stickyban_show()
+	if(!check_rights(R_BAN))
+		return
+		
+	var/list/bans = world.GetConfig("ban")
+	var/banhtml = ""
+	for(var/key in bans)
```
Comment (MrStonedOne) [inline code/modules/admin/stickyban.dm]: hey! listen!

So I just noticed that this means it's not sorted. I'm gonna put up a pr to sort the output on tg, (just run the bans var thru a sort.)

Just thought you should know.

Comment (Markolie) [inline code/modules/admin/stickyban.dm]: Ah, very nice. I'll port that once it's up. Thank you for the heads up!

Oh, and thank you for your hard work on this! We've been waiting for a system like this for quite a while!

**Comment by Fox-McCloud** [comment]
@Markolie 

https://github.com/tgstation/-tg-station/pull/12058

**Comment by Markolie** [comment]
Added.

**Comment by Alex6511** [comment]
Whats a sticky ban?

**Comment by Markolie** [comment]
See https://github.com/tgstation/-tg-station/pull/11205#issuecomment-131200519.

**Comment by MrStonedOne** [comment]
This is the post I left in our admin only forum. (as well as a memo pointing to it)

````
So byond has this thing, called a sticky ban, that is super good at catching ban evaders.

No idea how it does it, but it does it, it catches people using the cid randomizer, it catches people using vms, it even caught somebody on a new computer and new isp who hadn't connected for a year.

Some notes and rules:

Do not be excessive about using it. it can have false positives or go nuts and ban everybody. Only use it on people who are permabanned and unlikely to appeal said permaban, or people who are known to evade or claimed they will evade.

Stickybans are checked AFTER normal bans, so you can make the reason something snarky that refers to their failed evade.

Stickybans are synced between the servers, but this only updates between rounds, or when somebody opens the stickyban panel.

You can disassociate a ckey from a sticky ban, but 99% of the time, it will only get reassociated.

Admins are immune to stickybans created by this interface, but not stickybans created by headmins via remoting to the server.
````

**Comment by Markolie** [comment]
Thanks - I'll pass that on to our admins.

**Comment by Robertha89** [comment]
Can this actually be reverted? This is the second time me and a dozen others probably have been caught by the stickyban. At least untill the system bugs of banning a dozen other people by accident is fixed? If not, actually fix the part where it bans a dozen others.

**Comment by tigercat2000** [comment]
@Robertha89 That's not 'fixable';Stickybans are for the worst offenders, and they are managed entirely by BYOND's servers. False positives can happen, deal with it.

**Comment by Crazylemon64** [comment]
Rather, the presence of an admin during a sticky ban would likely facilitate fixing the false positives? At least, the last two times I was hit by the wide-angle disintegration beam, the admins were on point and fixed things up shortly after. The minichat is a good place to hang out if you're caught by the death laser. And as a non-admin, you can't really "deal with" a sticky ban, that being out of your own hands and all.

**Pull request merged**